### PR TITLE
Add test for optional input in Python backend

### PR DESCRIPTION
**Pull request by Tabrizian**

### CODE REVIEW

File: qa/L0_backend_python/python_test.py [hunk 1]
```diff
@@ -61,6 +61,56 @@ def _infer_help(self, model_name, shape, data_type):
...
+    def _optional_input_infer(self, model_name, input0, input1):
+        with httpclient.InferenceServerClient("localhost:8000") as client:
+            shape = (1,)
+            if input0:
+                input0_numpy = np.random.randint(0,
+                                                 100,
+                                                 size=shape,
+                                                 dtype=np.int32)
+            else:
+                # Set the input0 dummy data for testing purposes.
```
Comment (GuanLuo) [inline qa/L0_backend_python/python_test.py:73]: Not really "dummy" data, should say set to default input0 data that the model uses if not provided.

**Review by GuanLuo — DISMISSED** [review]

File: qa/L0_backend_python/python_test.py [hunk 2]
```diff
@@ -263,6 +315,14 @@ def test_unicode(self):
                     self.assertIsNotNone(output0)
                     self.assertEqual(output0[0], input_data)
 
+    def test_optional_input(self):
+        model_name = "optional"
+
+        with self._shm_leak_detector.Probe() as shm_probe:
+            for input0 in [True, False]:
```
Comment (tanmayv25) [inline qa/L0_backend_python/python_test.py:322]: input0/input1 -> skip_input0/skip_input1

The variable name should clarify that it is boolean. Note: there will be reversal in logic computation.

File: qa/python_models/optional/model.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import triton_python_backend_utils as pb_utils
+import numpy as np
+
+
+class TritonPythonModel:
+
```
Comment (tanmayv25) [inline qa/python_models/optional/model.py:32]: Describe about the model in few lines -- especially what it does when input is not provided?

**Review by tanmayv25 — CHANGES_REQUESTED** [review]

File: qa/python_models/optional/model.py (hunk 3 above)
Comment (tanmayv25) [reply qa/python_models/optional/model.py:32]: You can explain that in L34 of how does it support optional inputs.

Comment (Tabrizian) [reply qa/python_models/optional/model.py:32]: The model is self-explanatory but I added comment per your request.

**Review by tanmayv25 — APPROVED** [review]

**Review by GuanLuo — DISMISSED** [review]

**Comment by Tabrizian** [comment]
CI passed: [4456183](https://gitlab-master.nvidia.com/dl/dgx/tritonserver/-/pipelines/4456183) ✔️

**Review by GuanLuo — APPROVED** [review]

**Pull request merged**